### PR TITLE
postgresql version >=10 onwards does not require contrib package

### PR DESCRIPTION
**Pull request by KoenDierckx**
https://tickets.puppetlabs.com/browse/PDB-5454

### CODE REVIEW

File: manifests/database/postgresql.pp [hunk 1]
```diff
@@ -59,8 +59,11 @@
     # Only install pg_trgm extension, if database it is actually managed by the module
     if $manage_database {
 
-      # get the pg contrib to use pg_trgm extension
-      class { '::postgresql::server::contrib': }
+      # from postgresql version 10 onwards, this extension is no longer inside the contrib package, but is being bundled with the postgresql package itself
+      if (versioncmp($postgres_version, '10') == -1) {
```
Comment (smortex) [inline manifests/database/postgresql.pp:63]:
```suggestion
      if (versioncmp($postgres_version, '10') < 0) {
```

Better always compare to 0: this allows to read the condition as `$postgres_version < 10` by moving the operator between the parameters of `versioncmp` in your mind :wink:

File: manifests/database/postgresql.pp [hunk 2]
```diff
@@ -59,8 +59,11 @@
...
+        # get the pg contrib to use pg_trgm extension
+        class { '::postgresql::server::contrib': }
```
Comment (smortex) [inline manifests/database/postgresql.pp:65]: Maybe while here we can switch to:
```suggestion
        include postgresql::server::contrib
```
I remember having been beaten by this and reorder my profiles inclusions in a role when dealing with this module.

**Comment by KoenDierckx** [comment]
I signed to license (multiple times), but the check keeps coming back ?

**Comment by KoenDierckx** [comment]
License has been signed since 11 February
Is there something wrong with the check ?
Or am I missing something ?

image

**Comment by smortex** [comment]
Maybe commit [b53e922](https://github.com/puppetlabs/puppetlabs-puppetdb/pull/348/commits/b53e922531135f5b47370fd1435081a7d0f251ae) by GitHub in our names is causing trouble?

Maybe you can try to combine your commits in a single one on top of master?

From your working directory, assuming you are on commit [c19ba09](https://github.com/puppetlabs/puppetlabs-puppetdb/pull/348/commits/c19ba091fa75536541b173a55a8b011de0dedf94) and the origin remote point to this repository:

```
git rebase -i origin/main # Interactively rewrite history
```

This will bring your editor listing all commits in your branch.  Keep the first line unchanged, and replace `pick` with `squash` for all subsequent lines.  Save and quit, your editor will pop again and allow you to combine all the commit messages in a single one.

Then push your updated branch:

```
git push -f            # Send the changes (-f is required because we re-wrote history)
```

**Comment by KoenDierckx** [comment]
Tried your suggestions, did not help. I'll create a new PR

**Pull request closed without merging**